server/plugins/NoiseUGens.cpp

```diff
@@ -790,9 +790,13 @@ void RandID_Ctor(RandID* unit) {
 
 void RandID_next(RandID* unit, int inNumSamples) {
     float id = ZIN0(0);
+    bool fire = ZIN0(1) > 0;
```

Comment:
is there a way we can know the number of allocated UGen inputs, so we could write

```c
bool fire = num_ugen_args >= 2 && ZIN0(1) > 0
```
Then we would avoid crashing against clients and synth-defs using the previous ABI.

Reply:
Christoph noted that there _is_ already a `numInputs` that could/should be used here.

Reply:
How about this?

```
    bool fire = (unit->mNumInputs >= 2 ? ZIN0(1) : 0.f) > 0.f;
```

Reply:
Should it be
```c
bool fire = unit->mNumInputs >= 2 ? (ZIN0(1) > 0.f) : false;
```
?

Reply:
> `bool fire = unit->mNumInputs >= 2 && (ZIN0(1) > 0.f);`

OK, done, passes an inputs-mismatch test.